lib/config.php

```diff
@@ -357,7 +357,15 @@ private function new_settings($username) {
      */
     private function decrypt_settings($data, $key) {
         if (!$this->crypt) {
-            $data = $this->decode($data['settings']);
+            $settings = $data['settings'];
+            if (is_resource($settings)) {
```

Comment:
Can we do this at the place where stream is defined/used? It seems awkward to work with streams here. THis method should receive a string.